File: 2023-2024/sourcecode/mod2/Lab7/MVVMBasicsJava/app/src/main/java/com/example/mvvmbasicsjava/ui/fragment/UserViewHolder.java
package com.example.mvvmbasicsjava.ui.fragment;

import androidx.recyclerview.widget.RecyclerView;

import com.example.mvvmbasicsjava.databinding.RvItemBinding;
import com.example.mvvmbasicsjava.model.User;

public class UserViewHolder extends RecyclerView.ViewHolder {
    private RvItemBinding binding;

    public UserViewHolder(RvItemBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(User item) {
        String fullName = item.getFirstName() + " " + item.getLastName();
        binding.wordTextView.setText(fullName);
    }
}

File: 2023-2024/sourcecode/mod3/lab8/MVVMBasicsKotlin/app/src/main/java/com/example/mvvmbasicskotlin/ui/fragment/UserViewHolder.kt
package com.example.mvvmbasicskotlin.ui.fragment

import androidx.recyclerview.widget.RecyclerView
import com.example.mvvmbasicskotlin.databinding.RvItemBinding
import com.example.mvvmbasicskotlin.model.User

class UserViewHolder(private val binding: RvItemBinding) : RecyclerView.ViewHolder(binding.root) {
    fun bind(item: User) {
        val fullName = "${item.firstName} ${item.lastName}"
        binding.wordTextView.text = fullName
    }
}